wp-content/plugins/ig-revisions/plugin.php

```diff
@@ -49,3 +49,17 @@ function ig_revisions_metabox_save ( $post_id ) {
 add_action('publish_post', 'ig_revisions_metabox_save');
 add_action('edit_page_form', 'ig_revisions_metabox_save');
 
+function update_post_with_revision( $post ) {
+	$output_post = $post;
+	$revision_id = get_post_meta( $post->ID, 'ig_revision_id', true );
+	if(is_numeric($revision_id) && $revision_id >= 0) {
+		$revision_post = wp_get_post_revision( $revision_id );
+		$output_post = [
```

Comment:
It's now using the wp_api_extensions_output_post filter. There it's still an array and we do not have to do the conversion.

Reply:
Ok, but in this case we have to use `array_merge()` or overwrite only the single keys, otherwise the whole `$output_post`-array will be overwritten and the original keys as `id` etc. are missing.